internal/utilities/pruner/old_process_pruner.go

```diff
@@ -0,0 +1,118 @@
+package pruner
+
+import (
+	"context"
+	"fmt"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"time"
+	//lagoonv1beta1 "github.com/uselagoon/remote-controller/apis/lagoon/v1beta1"
+	//"github.com/uselagoon/remote-controller/internal/helpers"
+)
+
+// LagoonOldProcPruner will identify and remove any long running builds or tasks.
+func (p *Pruner) LagoonOldProcPruner(pruneBuilds, pruneTasks bool) {
+	opLog := ctrl.Log.WithName("utilities").WithName("LagoonOldProcPruner")
+	opLog.Info("Beginning marking old build and task pods")
+	namespaces := &corev1.NamespaceList{}
+	labelRequirements, _ := labels.NewRequirement("lagoon.sh/environmentType", selection.Exists, nil)
+	listOption := (&client.ListOptions{}).ApplyOptions([]client.ListOption{
+		client.MatchingLabelsSelector{
+			Selector: labels.NewSelector().Add(*labelRequirements),
+		},
+	})
+	if err := p.Client.List(context.Background(), namespaces, listOption); err != nil {
+		opLog.Error(err, fmt.Sprintf("Unable to list namespaces created by Lagoon, there may be none or something went wrong"))
+		return
+	}
+
+	//now we iterate through each namespace, and look for build/task pods
+	for _, ns := range namespaces.Items {
+		if ns.Status.Phase == corev1.NamespaceTerminating {
+			// if the namespace is terminating, don't search it for long running tasks
+			opLog.Info(fmt.Sprintf("Namespace %s is being terminated, skipping build/task pruner", ns.ObjectMeta.Name))
+			continue
+		}
+
+		podList := corev1.PodList{
+			TypeMeta: metav1.TypeMeta{},
+			ListMeta: metav1.ListMeta{},
+			Items:    nil,
+		}
+
+		listOption := (&client.ListOptions{}).ApplyOptions([]client.ListOption{
+			client.InNamespace(ns.ObjectMeta.Name),
+			client.MatchingLabels(map[string]string{
+				"lagoon.sh/controller": p.ControllerNamespace, // created by this controller
```

Comment:
We might be able to do a selector here that checks if the pod is in the running state too? That way the list of pods to iterate over will be less, and only relevant running pods returned.

Reply:
Good call!

Reply:
Not sure if we have a buildstatus/taskstatus label, pretty sure we don't, so could be worth using the `lagoon.sh/jobType` label to check if it just exists

The `lagoon.sh/controller` label is pretty good at covering most things though. So don't feel like you need to solve this, was more of a nice to maybe do if simple enough.

Reply:
Yeah looking at this with more eyes. I think if we just tack in a `lagoon.sh/jobType` label existence check (like this https://github.com/uselagoon/remote-controller/pull/183/files#diff-690a4e7b25632837a77f96ac1668f15b0bb494b8a750e021e73cd43d83380778R22) that should cover off so that the only pods selected are builds/tasks.

This will return pods that are running, complete or failed if they exist in the cluster, but then the follow up status check you're doing will only execute on running pods so we're good on the running only front